packages/amplify-cli-core/src/feature-flags/featureFlagEnvironmentProvider.ts

```diff
@@ -0,0 +1,148 @@
+import * as path from 'path';
+import { config as dotenvConfig } from 'dotenv';
+
+import { EnvVarFormatError, FeatureFlags } from '.';
+import { FeatureFlagValueProvider } from './featureFlagValueProvider';
+
+export type FeatureFlagEnvironmentProviderOptions = {
+  prefix?: string;
+  environmentNameSeparator?: string;
+  noEnvironmentNameSeparator?: string;
+  envPathSeparator?: string;
+  internalSeparator?: string;
+  projectPath?: string;
+};
+
+const defaultFeatureFlagEnvironmentProviderOptions = {
+  prefix: 'AMPLIFYCLI',
+  environmentNameSeparator: '-',
+  noEnvironmentNameSeparator: '_',
+  envPathSeparator: '__',
+  internalSeparator: ':',
+};
+
+/**
+ * Using the environment variables of the current process and a .env file to transform those into feature flags values.
+ * The resulting object has a 'project' member and an 'environments' member. 'project' contains the project level feature flags,
+ * 'environments' contains objects, each keyed by the name of the environment the flags are defined for.
+ *
+ * Environment variables must have this format:
+ * {prefix}_{section}__{property} or {prefix}-{CLI environment name}_{section}__{property}
+ */
+export class FeatureFlagEnvironmentProvider implements FeatureFlagValueProvider {
+  constructor(private options: FeatureFlagEnvironmentProviderOptions = defaultFeatureFlagEnvironmentProviderOptions) {
```

Comment:
I am wondering if we can move the `transformEnvFlags` method to this class as we dont need to transform them when they come from a JSON file.

Reply:
Nope, transform requires the registration data and the providers are not coupled to the behavior of  `FeatureFlagProvider` itself. They preserve whatever was read.